infra/azure/modules/auth/main.tf

```diff
@@ -21,6 +21,27 @@ module "auth_sp" {
   application_object_id = azuread_application.auth.object_id
 }
 
+resource "azuread_application" "test_auth" {
+  display_name = "${var.resource_group_name}-test-auth"
+
+  required_resource_access {
+    resource_app_id = "00000003-0000-0000-c000-000000000000"
+
+    resource_access {
+      # Application.ReadWrite.All
+      id   = "1bfefb4e-e0b5-418b-a88f-73c46d2cc8e9"
+      type = "Role"
+    }
```

Comment:
Is the permissions/roles grant for auth? I was expecting an `azurerm_role_assignment` that granted it, I dunno, azure app creation/deletion/sp-creation.

Reply:
Ya there's a different way of assigning permissions within AAD vs permissions on other resources in Azure. This was the only way I could find at the time to create/read apps and service principals.